### PR TITLE
#89 - Required Keys & #90 - Optional Keys

### PR DESCRIPTION
**Pull request by yi-tuan**
Signed-off-by: liuxing.yituan <liuxing.yituan@bytedance.com>

### CODE REVIEW

**Comment by antfu** [comment]
@yi-tuan Thanks for the new challenges! They look good to me, however, please use the "New Challenge" issue next time, it will auto-generate PRs.

PS. Your commit email does not seem to linked with your Github account. Just be sure if it's desired, let me know if you feel it's ready to be merged.

File: questions/89-hard-required-keys/README.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<!--info-header-start--><h1>Required Keys <img src="https://img.shields.io/badge/-hard-red" alt="hard"/> <img src="https://img.shields.io/badge/-%23utils-999" alt="#utils"/> </h1><blockquote><p>yituan <a href="https://github.com/yi-tuan" target="_blank">@yi-tuan</a></p></blockquote><p><a href="https://tsch.js.org/89/play" target="_blank"><img src="https://img.shields.io/badge/-Take%20the%20Challenge-3178c6?logo=typescript" alt="Take the Challenge"/></a> </p><!--info-header-end-->
+
+Implement the advanced util type `RequiredKeys<T>`, which picks all the required keys
```
Comment (antfu) [inline questions/89-hard-required-keys/README.md:3]: Should this be a union if multiple required keys present? If so, I guess to better to provide some examples & test cases about that. Same as #91. Thanks

**Comment by antfu** [comment]
Ah, I see. The action is broken, will let you know when it's fixed. And let's utilize issues to discuss.

**Comment by yi-tuan** [comment]
Ok～  Let me make a change and mention PR again.

**Comment by yi-tuan** [comment]
> fixed



> Ah, I see. The action is broken, will let you know when it's fixed. And let's utilize issues to discuss.

Do I need to create "New Challenge" issue  issue again？

**Comment by antfu** [comment]
> Do I need to create "New Challenge" issue again？

No, modifying the issue will trigger the rebuild.

**Comment by yi-tuan** [comment]
> > Do I need to create "New Challenge" issue again？
> 
> No, modifying the issue will trigger the rebuild.

please let me know when it's fixed.

**Pull request closed without merging**